app/app_abci.go

```diff
@@ -53,7 +54,8 @@ func (app *OKExChainApp) DeliverRealTx(req abci.TxEssentials) (res abci.Response
 		}
 		if err == nil {
 			//optimize get tx gas price can not get value from verifySign method
-			app.blockGasPrice = append(app.blockGasPrice, tx.GetGasPrice())
+			app.gpo.CurrentBlockGPs.AddGP(tx.GetGasPrice())
+			app.gpo.CurrentBlockGPs.AddGas(tx.GetGas())
```

Comment:
If AddGP and AddGas cannot be used alone, maybe it would be better to put them in a function like Update(gp, gas)

Reply:
down